### PR TITLE
AHOYAPPS-371: Onboarding Implementation

### PR DESCRIPTION
**Pull request by Alton09**
## Description

Added happy path community login integration tests and began implementation of detailed error handling for Twilio auth function.

## Breakdown

- Add happy path integration tests for CommunityLoginActivity.
- Add data classes and custom exception for Twilio auth function error handling.

## Validation

- Passed manual validation and CI pipeline.

## Additional Notes

N/A

## Submission Checklist

 - [x] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -190,6 +190,7 @@ dependencies {
     testImplementation 'junit:junit:4.12'
     testImplementation 'com.nhaarman.mockitokotlin2:mockito-kotlin:2.2.0'
     testImplementation 'org.robolectric:robolectric:4.3'
+    testImplementation 'pl.pragmatists:JUnitParams:1.1.1'
```
Comment (aaalaniz) [inline app/build.gradle]: Love this library

File: app/src/community/java/com/twilio/video/app/data/api/AuthServiceRepository.kt [hunk 2]
```diff
@@ -16,31 +16,43 @@
...
                 val response = authService.getToken(url, requestBody)
                 Timber.d("Token returned from Twilio auth service: %s", response)
                 return response.token!!
+            } catch (e: HttpException) {
+                // TODO Handle all error scenarios as part of https://issues.corp.twilio.com/browse/AHOYAPPS-446
+                val errorJson = e.response()!!.errorBody()!!.string()
+                val errorDTO = Gson().fromJson(errorJson, AuthServiceErrorDTO::class.java)
+                Timber.e(e, errorDTO.error!!.explanation)
+                val error = AuthServiceError.value(errorDTO.error.message!!)
+                throw AuthServiceException(e, error)
```
Comment (aaalaniz) [inline app/src/community/java/com/twilio/video/app/data/api/AuthServiceRepository.kt]: This threw me for a sec, but this is part of the co-routine adoption 😅

File: app/src/main/java/com/twilio/video/app/data/api/TokenService.kt [hunk 3]
```diff
@@ -17,5 +17,5 @@
 package com.twilio.video.app.data.api
 
 interface TokenService {
-    suspend fun getToken(identity: String? = null, roomName: String? = null): String
+    suspend fun getToken(identity: String? = null, roomName: String? = null, passcode: String? = null): String
```
Comment (aaalaniz) [inline app/src/main/java/com/twilio/video/app/data/api/TokenService.kt]: Can we keep two `getToken` declarations and make one optional? This way, we don't need the non-community flavors implement a method where the passcode is not used.

File: app/src/main/java/com/twilio/video/app/data/api/VideoAppServiceDelegate.kt [hunk 4]
```diff
@@ -32,7 +32,7 @@ class VideoAppServiceDelegate(
     private val videoAppServiceProd: VideoAppService
 ) : TokenService {
 
-    override suspend fun getToken(identity: String?, roomName: String?): String {
+    override suspend fun getToken(identity: String?, roomName: String?, passcode: String?): String {
```
Comment (aaalaniz) [inline app/src/main/java/com/twilio/video/app/data/api/VideoAppServiceDelegate.kt]: See my comment above, but I think we should update the interface definition so the passcode is not referenced outside the community version

File: app/build.gradle (hunk 1 above)
Comment (Alton09) [reply app/build.gradle]: It's awesome!

**Pull request merged**